inference-engine/src/vpu/graph_transformer/src/stages/dynamic_shape_resolver.cpp

```diff
@@ -19,6 +19,9 @@ void FrontEnd::parseDSR(const Model& model, const ie::CNNLayerPtr& layer, const
     const auto dataProducerEdge = data->producerEdge();
     VPU_THROW_UNLESS(dataProducerEdge != nullptr, "Parsing layer {} of type {} failed: input with index {} (of name {}) must have a producer",
         layer->name, layer->type, 0, data->name());
+    VPU_THROW_UNLESS(data->consumerEdges().size() == 0,
+        "Parsing layer {} of type {} failed: input with index {} (of name {}) must have no consumers, actual: {}",
+        layer->name, layer->type, 0, data->name(), data->consumerEdges().size());
```

Comment:
Nitpick: you can make assertion message more friendly. It seems to me there is some semantic (like consumers could not be parsed before or their existence means we try to process garbage data or kind of that) under assertion, you can express it directly in message, so when someone reads code in the future can avoid thinking about the root cause for an assertion.

Reply:
Done. You can check it out in amended commit